### ISSUE TITLE
Add comment to "Resize" that "resets" pane sizes

### ISSUE THREAD

**Issue by macintacos** [zellij-org/zellij#1289]
I expected there to be a "Reset" command (or something like it) under the "Resize" parent command that would essentially evenly spread out the available panes. Doesn't seem like that's possible. Am I missing something, or is this not possible right now?

**Comment by jaeheonji** [comment]
Good point! Yes, it is currently not available. but, after executing resize action (`Ctrl+n` by default), for example, if you press `<ESC>`, the ability to return to its original size seems really necessary!
Thank you for your good feedback 👍

**Comment by raphCode** [comment]
> ability to return to its original size seems really necessary

While this sounds good too, I think OP wanted a different behavior: Splitting the available space evenly between panes, like in vim with `Ctrl-W =`

**Comment by raphCode** [comment]
Since spilts are nested hierarchically, it could be useful to have a command "evenly resize only this and child splits" or "evenly resize only this split"

For example, any of these command on the left part of the vertical split
```
┌─────┬───┐
├─────┤   │
│     │   │
│     │   │
└─────┴───┘
```
results in this layout, keeping the vertical split at its position / not resizing the big right pane:
```
┌─────┬───┐
│     │   │
├─────┤   │
│     │   │
└─────┴───┘
```
Resizing all splits would lead to:
```
┌────┬────┐
│    │    │
├────┤    │
│    │    │
└────┴────┘
```
This seems mostly a UI problem, since we currently only can focus single panes and not splits which consist of multiple panes.

**Comment by imsnif** [comment]
Hey, just a note: splits are not nested hierarchically in Zellij. This was a deliberate design choice to allow for much greater freedom when resizing as opposed to every other (terminal) multiplexer I know of. You can see this freedom by splitting the workspace into four panes (split right, split down, change focus left, split down) and then trying to resize any pane in any direction. In Zellij this is possible, in other multiplexers it is not.

This also means we can solve this problem by breaking panes out of their nestings. I.e. if we have 8 panes on screen we could make all of them equal sized. But honestly - ever since I've seen this issue I've been wondering: what is the intent here? I mean, if I have panes only split horizontally or only split vertically: this feature makes total sense. But if I have a more complex setup - what is this command to do? What does the user want here?

**Comment by jaeheonji** [comment]
First, I think we need to seprate the two idea.

My first idea is to implement a cancel action in the RESIZE.
Second, The idea of ​​@raphCode, for example, can be thought of as making the size of all pane or pane of a specific area the same ratio. And maybe?, this will make all the pans on the current screen a Grid with the same size.

I think both are pretty good ideas. And it seems that the latter is the intention of @macintacos. Is it right?

**Comment by raphCode** [comment]
> splits are not nested hierarchically in Zellij

I guess they are not implemented like that because of the demo you mentioned.
But from a layout configuration it indeed looks like a hierarchy.

> But if I have a more complex setup - what is this command to do? What does the user want here?

I don't have a definite answer to this but I went with the most simple case: every split equalizes the size of its "children".

**Comment by macintacos** [comment]
What @raphCode said [here](https://github.com/zellij-org/zellij/issues/1289#issuecomment-1087598404) is pretty much exactly what I want.

**Comment by imsnif** [comment]
About hierarchical nesting - there are some setups in which (at least to my understanding) it's not trivial (or even possible?) to find out what the children are. Here's an example:
![img-2022-04-05-163622](https://user-images.githubusercontent.com/795598/161779220-9880eb10-a032-4af6-a155-12afa0d89399.png)
(apologies for the similarity to some mid-20th-century-regime-who-killed-my-grandparents symbolism - it's just about the geometry here)

I'm not saying the above suggestion wouldn't be possible in most cases, but it's important to also mind these.

**Comment by raphCode** [comment]
Now I am intrigued - how do you even create such a layout?

(Maybe such "crazy layouts" would be nice to have in the docs/screenshots, or just for a show-off compared to other terminal multiplexers)

**Comment by imsnif** [comment]
> Now I am intrigued - how do you even create such a layout?

A bit involved, but:
start zellij
split down
change focus up
split down
split right
change focus left
split right
change focus up
split right
resize up
close pane
change focus down
split right
change focus left
split right
change focus left
resize up
close pane
change focus right
change focus right
resize right
close pane
 
> (Maybe such "crazy layouts" would be nice to have in the docs/screenshots, or just for a show-off compared to other terminal multiplexers)

I kind of tried to explain this in the beta announcement a year ago, but I think I didn't do a very good job of it: https://zellij.dev/news/beta/

If you have a better idea of how to communicate this, that would be cool :)

**Comment by raphCode** [comment]
> A bit involved, but:

I see. The resizing steps are crucial, and for these to work in your workflow the window size must be right so that the splits align...

Alternative steps with also no window size requirements:
- create a 3x3 layout with the splits aligned (unequal pane sizes are fine)
- resize the panes directly touching the center pane (von Neumann neighborhood :nerd_face:). This "locks up" some of the split lines:
  - top one left
  - left one down
  - bottom one right
  - right one up
- close the panes in the 4 corners

Haha, I wondered what the layout engine does upon closing the center pane in that layout, and got different results in different tries:
- a crash:
```
Error occurred in server:

  × Thread 'screen' panicked.
  ├─▶ Originating Thread(s)
  │   	1. stdin_handler_thread: AcceptInput
  │   	2. screen_thread: WriteCharacter
  │
  ├─▶ At zellij-server/src/tab/mod.rs:826:60
  ╰─▶ called `Option::unwrap()` on a `None` value
```
- 
- 
- on the first try and a messed up display on the second one

> I think I didn't do a very good job of it

You made very neat graphics that describe 
That resizing in different directions is possible is

**Comment by oredaze** [comment]
With a risk of being annoying I will point to a reference - tmux has this feature, it's `select-layout -E`
Although, it may or may not be implementable in zellij (in that form).
